domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -1,32 +1,91 @@
 package org.oppia.domain.profile
 
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Matrix
+import android.media.ThumbnailUtils
 import android.net.Uri
+import android.provider.MediaStore
+import androidx.exifinterface.media.ExifInterface
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import org.oppia.app.model.Profile
+import org.oppia.app.model.ProfileDatabase
 import org.oppia.app.model.ProfileId
 import org.oppia.data.persistence.PersistentCacheStore
 import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
 import org.oppia.util.logging.Logger
+import org.oppia.util.profile.DirectoryManagementUtil
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.lang.Exception
+import java.math.BigInteger
+import java.nio.charset.Charset
+import java.security.MessageDigest
+import java.security.NoSuchAlgorithmException
+import java.util.*
 import javax.inject.Inject
 import javax.inject.Singleton
 
+private const val TRANSFORMED_GET_PROFILES_PROVIDER_ID = "transformed_get_profiles_provider_id"
+private const val TRANSFORMED_GET_PROFILE_PROVIDER_ID = "transformed_get_profile_provider_id"
+private const val GRAVATAR_URL_PREFIX = "https://www.gravatar.com/avatar/"
+private const val GRAVATAR_QUERY_STRING = "?s=100&d=identicon&r=g"
+
 /** Controller for retrieving, adding, updating, and deleting profiles. */
 @Singleton
 class ProfileManagementController @Inject constructor(
   private val logger: Logger,
-  private val persistentCacheStoreFactory: PersistentCacheStore.Factory
-){
+  cacheStoreFactory: PersistentCacheStore.Factory,
+  private val dataProviders: DataProviders,
+  private val context: Context,
+  private val directoryManagementUtil: DirectoryManagementUtil
+) {
   private var currentProfileId: Int = -1
+  private val profileDataStore = cacheStoreFactory.create("profile_database", ProfileDatabase.getDefaultInstance())
+
+  class ProfileNameNotUniqueException(msg: String): Exception(msg)
+  class ProfileNameOnlyLettersException(msg: String): Exception(msg)
+  class FailedToStoreImageException(msg: String): Exception(msg)
+  class FailedToDeleteDirException(msg: String): Exception(msg)
+  class ProfileNotFoundException(msg: String): Exception(msg)
+  class FailedToReadProfilesException(msg: String): Exception(msg)
+
+  enum class ProfileActionStatus {
+    Success,
```

Comment:
Enum constants should be fully capitalized per https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names.

Reply:
Updated to use CONSTANT_CASE.